### DISCUSSION TITLE
How to extend the project to allow local dev?

### DISCUSSION THREAD

**Discussion by takeda-ma** [aws-samples/websocket-chat-application#57] [Q&A]
How do we local dev? I can see that we are running few lambdas for rest and websocket (I think I can somehow replicate this in local dev by running node server) but how about the cognito and other stuff I am probably missing?

**Comment by santatamas** [comment]
Hi! It's possible to local dev the UI. You can also run Lambdas locally and run a local DynamoDB container. Emulating the rest of the AWS services is not possible at the moment.

**Reply by takeda-ma** [reply]
@santatamas hey thanks for the reply.
What do you think are the advantage of using api gateway + lambda for chat app versus just using aws fargate as explained in this [article](https://medium.com/containers-on-aws/building-a-socket-io-chat-app-and-deploying-it-using-aws-fargate-86fd7cbce13f)

**Reply by santatamas** [reply]
The advantage is the low operational overhead, scalability and the fact that you basically pay nothing if noone uses your app. So for smaller chat apps I could recommend this solution. For large-scale projects, either use ECS + Fargate, or AppSync. IoT Core with MQTT over websockets could also be an option.
